classification/apply_HLDA.m

```diff
@@ -0,0 +1,61 @@
+function out = apply_HLDA(epo, ival, C)
+% TRAIN_LDA - linear discriminant analysis 
+%
+%Synopsis:
+%   out = apply_HLDA(epo, ival, C)
+% 
+%Arguments:
+%   epo: STRUCT         - Structure containing epoched ERP data, with
+%                           obligatory fields 'x', 'y' and 't'.
+%   ival: INT [2]       - interval of the ERP on which the training should be
+%                           performed. Must match the
+%                           size of the interval used to train the
+%                           classifier C.
+%   C: STRUCT           - a hierarchical LDA classifier structure that 
+%                           contains the individual segment classifiers and 
+%                           the top-level classifier. Must include the
+%                           fields 'seg' and 'final'
+%Returns:
+%   out: FLOAT[]        - an array containing the classifier score for each 
+%                           sample
+%Description:
+%   APPLY_HLDA applies a hierarchical LDA classifier given an epo structure,
+%               a training interval and a number of segments
+%
+%Examples:
+%   apply_HLDA(epo, ival, C))
+%   
+%See also:
+%   TRAIN_HLDA
+
+misc_checkType(epo, 'STRUCT(x y t)');
+misc_checkType(ival, 'INT[2]');
+misc_checkType(C, 'STRUCT(seg final)');
+
+
+nSegments = length(C.seg);
+dims = size(epo.x);
+% convert interval values to indices
+ival_idx = find(epo.t == ival(1)) : find(epo.t == ival(2));

```

Comment:
This is a bit problematic. epo.t might not include the exact values given (think, e.g., about the case of a sampling rate of 256Hz). Therefore you should do something like:
     ibeg= max([1 find(epo.t<=ival(1))]);
     iend= min([find(epo.t>=ival(2)) length(epo.t)]);
     ival_idx= ibeg:iend;
However, you could also use the function proc_selectIval.